### PR TITLE
changed all asset to if/raise to prevent disable of assert during PYTHONOPTIMISE env

### PR DESCRIPTION
**Pull request by ViveK-PothinA**
## Description
Changed all assert blocks to if/raise block. This is important as assert checks can be globally disabled with the -O and -OO flags, as well as the PYTHONOPTIMIZE env var. This PR is against an open issue, [Don't use `assert` for non-debugging data validation](https://github.com/OpenMined/PySyft/issues/3480) 

## Affected Dependencies
For now, commented out the assert blocks and wherever error statement wasn't present, added an error statement based on understanding I could make of it 

## How has this been tested?
- ran the standard `python setup.py test`, which was a success

### CODE REVIEW

**Comment by tudorcebere** [comment]
Really love this, looking forward to see it merged! Congrats for your first PR @ PySyft!

**Comment by ViveK-PothinA** [comment]
> Really love this, looking forward to see it merged! Congrats for your first PR @ PySyft!

Thanks @tudorcebere , my first PR in open source world 😬 , hopefully will do more going forward.

**Comment by aanurraj** [comment]
Nice work with coverage :)

File: syft/frameworks/crypten/__init__.py [hunk 1]
```diff
@@ -86,7 +90,12 @@ def load_model(tag: str):
     results = worker.search(tag)
 
     # Make sure there is only one result
-    assert len(results) == 1
+    # assert len(results) == 1
+    if len(results) != 1:
+        if len(results) == 0:
+            raise RuntimeError("No worker result found.")
```
Comment (gmuraru) [inline syft/frameworks/crypten/__init__.py:96]: Q: Could you thorw dirrectly?
```raise RuntimeError(f"Number of results found {len(results}}. Expected 1")

File: syft/frameworks/crypten/message_handler.py [hunk 2]
```diff
@@ -44,12 +44,16 @@ def run_crypten_party_plan(self, msg: CryptenInitPlan) -> ObjectMessage:  # prag
 
         # TODO Change this, we need a way to handle multiple plan definitions
         plans = self.worker.search("crypten_plan")
-        assert len(plans) == 1
+        # assert len(plans) == 1
+        if len(plans) != 1:
+            raise ValueError("More than one plans found. Currently not supported.")
```
Comment (gmuraru) [inline syft/frameworks/crypten/message_handler.py:49]: Q: Also here

File: syft/frameworks/torch/nn/functional.py [hunk 3]
```diff
@@ -85,8 +89,11 @@ def _pre_conv(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=
     Because all the computation are local, we add the @allow_command and run it directly
     on each share of the additive sharing tensor, when running mpc computations
     """
-    assert len(input.shape) == 4
-    assert len(weight.shape) == 4
+
+    # assert len(input.shape) == 4
+    # assert len(weight.shape) == 4
+    if len(input.shape) != 4 or len(weight.shape) != 4:
```
Comment (gmuraru) [inline syft/frameworks/torch/nn/functional.py:95]: Q: Could we have 2 ```if``` statements here?

File: syft/frameworks/torch/nn/functional.py [hunk 4]
```diff
@@ -98,12 +105,22 @@ def _pre_conv(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=
...
 
     # Check if inputs are coherent
-    assert nb_channels_in == nb_channels_kernel * groups
-    assert nb_channels_in % groups == 0
-    assert nb_channels_out % groups == 0
+    # assert nb_channels_in == nb_channels_kernel * groups
+    # assert nb_channels_in % groups == 0
+    # assert nb_channels_out % groups == 0
+    if nb_channels_in != nb_channels_kernel * groups:
+        raise ValueError("Given inputs are not supported")
```
Comment (gmuraru) [inline syft/frameworks/torch/nn/functional.py:117]: Q: Could you specify here given inputs are not supported for ```nb_channels_in``` - the same for the ones bellow

File: syft/frameworks/torch/nn/functional.py [hunk 5]
```diff
@@ -220,8 +237,12 @@ def conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1):
...
 
     if isinstance(input.child, FrameworkTensor) or isinstance(weight.child, FrameworkTensor):
-        assert isinstance(input.child, FrameworkTensor)
-        assert isinstance(weight.child, FrameworkTensor)
+        # assert isinstance(input.child, FrameworkTensor)
+        # assert isinstance(weight.child, FrameworkTensor)
+        if not isinstance(input.child, FrameworkTensor):
+            raise TypeError("input needs to be FrameworkTensor")
+        if not isinstance(weight.child, FrameworkTensor):
+            raise TypeError("input needs to be FrameworkTensor")
```
Comment (gmuraru) [inline syft/frameworks/torch/nn/functional.py:245]: Q: Could you specify here that ```weight```, not ```input```

File: syft/frameworks/torch/nn/functional.py [hunk 6]
```diff
@@ -337,8 +360,12 @@ def _pre_pool(input, kernel_size, stride=1, padding=0, dilation=1, groups=1):
 
     # Check if inputs are coherent
     # assert nb_channels_in == nb_channels_kernel * groups
-    assert nb_channels_in % groups == 0
-    assert nb_channels_out % groups == 0
+    # assert nb_channels_in % groups == 0
+    # assert nb_channels_out % groups == 0
+    if nb_channels_in % groups != 0:
+        raise ValueError("Given inputs are not supported")
```
Comment (gmuraru) [inline syft/frameworks/torch/nn/functional.py:366]: Q: Also here, specify ```nb_channels_in``` (not the value, but the name of the variable)

File: syft/frameworks/torch/nn/functional.py [hunk 7]
```diff
@@ -337,8 +360,12 @@ def _pre_pool(input, kernel_size, stride=1, padding=0, dilation=1, groups=1):
...
+    if nb_channels_out % groups != 0:
+        raise ValueError("Given inputs are not supported")
```
Comment (gmuraru) [inline syft/frameworks/torch/nn/functional.py:368]: Q: Also here

File: syft/frameworks/torch/nn/functional.py [hunk 8]
```diff
@@ -527,11 +558,21 @@ def max_half_split(tensor4d, half_size):
...
-        assert output_size[0] == output_size[1]
+        # assert output_size[0] == output_size[1]
+        if output_size[0] != output_size[1]:
+            raise ValueError("Check given output_size")
         output_size = output_size[0]
-    assert tensor.shape[2] == tensor.shape[3]
+
+    # assert tensor.shape[2] == tensor.shape[3]
+    if tensor.shape[2] != tensor.shape[3]:
+        raise ValueError("Check given tensor")
```
Comment (gmuraru) [inline syft/frameworks/torch/nn/functional.py:568]: Q: given tensor shape

File: syft/frameworks/torch/nn/pool.py [hunk 9]
```diff
@@ -36,10 +36,17 @@ def __init__(
...
 
-        assert padding == 0
-        assert ceil_mode is False
-        assert count_include_pad is True
-        assert divisor_override is None
+        # assert padding == 0
+        # assert ceil_mode is False
+        # assert count_include_pad is True
+        # assert divisor_override is None
+        if (
```
Comment (gmuraru) [inline syft/frameworks/torch/nn/pool.py:43]: Q: Could you rewrite this with an
```if not (padding == 0 and ceil_mode...) <--- it shows that we support only one setting

File: syft/frameworks/torch/tensors/interpreters/additive_shared.py [hunk 10]
```diff
@@ -477,7 +477,11 @@ def _basic_arithmetic_op(self, op, shares: dict, operand):
                 *self.child.keys(), **self.get_class_attributes(), **no_wrap
             ).child
 
-        assert len(shares) == len(operand)
+        if len(shares) != len(operand):
+            raise ValueError(
+                "Size of shares(%d) is not equal to that of operand(%d)."
```
Comment (gmuraru) [inline syft/frameworks/torch/tensors/interpreters/additive_shared.py:482]: Q: Why not use f-string?

File: syft/frameworks/crypten/__init__.py (hunk 1 above)
Comment (ViveK-PothinA) [reply syft/frameworks/crypten/__init__.py:96]: sure

**Review by gmuraru — COMMENTED** [review]
LGTM! Only one last thing. Could you remove the comments with the asserts?

**Review by aanurraj — APPROVED** [review]

**Comment by ViveK-PothinA** [comment]
removed the commented assert statements.

**Review by gmuraru — APPROVED** [review]
LGTM. Thank you :D

**Pull request merged**